lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -594,35 +598,40 @@ sub variant_exists {
     }
   }
 
-  # Fix bug - if location matches but alleles don't - it gives bad dataset response
-
   if ($incl_ds_response) {
     my %datasets;
     # HIT - returns only datasets that have the queried variant
     # If has a list of datasets to query and a variant was found then print dataset response
-    if ($incl_ds_response == 2 && $has_dataset && $vf_found) {
-      foreach my $dataset_id (keys %dataset_var_found) {
-        if (exists $variation_set_list{$dataset_id}){
+    if ($incl_ds_response == 2 && $has_dataset == 1 && $vf_found) {
```

Comment:
`$has_dataset` is a boolean, so this was actually better before the last change!